config/src/providers/remappings.rs

```diff
@@ -47,8 +47,15 @@ impl<'a> RemappingsProvider<'a> {
         trace!("get all remappings from {:?}", self.root);
         /// prioritizes remappings that are closer: shorter `path`
         ///   - ("a", "1/2") over ("a", "1/2/3")
-        fn insert_closest(mappings: &mut HashMap<String, PathBuf>, key: String, path: PathBuf) {
-            match mappings.entry(key) {
+        /// grouped by remapping context
+        fn insert_closest(
+            mappings: &mut HashMap<Option<String>, HashMap<String, PathBuf>>,
```

Comment:
I think this should work,
so the previous hashmap is now just the `None` key

and this also supports remappings with context extracted from nested deps

Reply:
fixed in https://github.com/foundry-rs/foundry/pull/5397/commits/d92bd235fa551d073b7bfb09a3011a7d264387f5 — we can switch to master deps whenever we plan on publishing crates 😄